x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/geo_view/metric_selection.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useContext, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiSpacer } from '@elastic/eui';
+import { LayerDescriptor } from '@kbn/maps-plugin/common';
+
+import { JobCreatorContext } from '../../../job_creator_context';
+import { GeoJobCreator } from '../../../../../common/job_creator';
+import { GeoField } from './geo_field';
+import { GeoMapExamples } from './geo_map_examples';
+import { useMlKibana } from '../../../../../../../contexts/kibana';
+
+interface Props {
+  setIsValid: (na: boolean) => void;
+}
+
+export const GeoDetector: FC<Props> = ({ setIsValid }) => {
+  const { jobCreator: jc, jobCreatorUpdated, chartLoader } = useContext(JobCreatorContext);
+  const jobCreator = jc as GeoJobCreator;
+
+  const [fieldValues, setFieldValues] = useState<string[]>([]);
+  const [layerList, setLayerList] = useState<LayerDescriptor[]>([]);
+
+  const {
+    services: { data, notifications: toasts },
+  } = useMlKibana();
+  const { mapLoader } = useContext(JobCreatorContext);
+
+  useEffect(() => {
+    let valid = false;
+    if (jobCreator.geoField !== null) {
+      valid = true;
+    }
+    setIsValid(valid);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [jobCreatorUpdated, jobCreator.geoField]);
```

Comment:
if we're watching `jobCreatorUpdated` we shouldn't need to also watch `jobCreator.geoField`.
I know this seems counter intuitive, as really we should be watching all dependencies, but these wizards are designed around `jobCreatorUpdated` being the single notification that changes have happened.
So for consistency with the other components, I think we should only watch `jobCreatorUpdated` here.

Reply:
👌 Thanks for the context - updated in 9be34fed1d464e8fac17090c2aa5e2e78099e150